server/src/main/java/org/elasticsearch/search/aggregations/support/AggregationContext.java

```diff
@@ -63,7 +66,7 @@
  * this is {@code abstract} so that tests can build it without creating the
  * massing {@link QueryShardContext}.
  */
-public abstract class AggregationContext {
+public abstract class AggregationContext implements Releasable {
```

Comment:
Making `AggregationContext` `Releasable` lets us free the preallocated breaker after freeing all of the aggs without adding complexity to `SearchContext`.